pkg/kinflate/resource/resourcecollection.go

```diff
@@ -0,0 +1,41 @@
+package resource
+
+import (
+	"fmt"
+
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	"k8s.io/kubectl/pkg/kinflate/types"
+	"k8s.io/kubectl/pkg/loader"
+)
+
+type ResourceCollection types.KObject
```

Comment:
I don't understand why we have both `ResourceCollection` and `types.KObject`.
Also I don't really see the value of these two `New` function. They just tie things together that don't necessarly have to be tied (iow, RC <-> manifest, RC <-> patch) but the only added value is converting a list of resources into a ResourceCollection (the real value is in the private method).

What about this:
```
func NewKobjectFromResources(l ...*Resources) (types.KObject, error)
```
as an interface?

Reply:
> type ResourceCollection types.KObject

This is a intermediate state. After some cleanup in a followup PR, KObject will be gone and only ResourceCollection will be left.

Reply:
Oh, great!

Reply:
I think this line should be moved to the types pkg which will make it closer to its final state.